src/test/java/com/artipie/asto/KeyTest.java

```diff
@@ -48,4 +50,21 @@ void resovlesKeyFromBasePath() {
             Matchers.equalTo("black/red/green/yellow")
         );
     }
+
+    @Test
+    void keyFromString() {
+        final String string = "a/b/c";
+        MatcherAssert.assertThat(
+            new Key.From(string).string(),
+            Matchers.equalTo(string)
+        );
+    }
+
+    @Test
+    void keyFromBadParts() {
+        Assertions.assertThrows(
+            Exception.class,
+            () -> new Key.From("part1", "part/2").string()
+        );
```

Comment:
@olegmoz I'd suggest to improve this assertion to check also exception class and message, smth like
```
MatcherAssert.assertThat(
            Assertions.assertThrows(
                IllegalStateException.class, 
                () -> new Key.From("part1", "part/2").string()
           ).getMessage(),
           new IsEqual<>("exception message")
)
```

Reply:
@olenagerasimova sorry, but I do not see particular exception class or message to be part of method's contract right now, so I'd prefer to not enforce them in unit test

Reply:
@olegmoz hm, you are talking about `Key#string()` method contract, right? This method does not announce any exception in its contract, but you still wrote this test... Also, don't we write unit tests for particular implementations? The implementation of the `Key` this test is about throws `IllegalStateException` on line 128, why not to specifically test it?

Reply:
@olenagerasimova added checks to test